src/xunit.analyzers/InlineDataMustMatchTheoryParameters.cs

```diff
@@ -229,5 +208,65 @@ internal enum ParameterArrayStyleType
             /// </summary>
             Initializer,
         }
+
+        private static class ConversionChecker
+        {
+            public static bool IsConvertible(Compilation compilation, ITypeSymbol source, ITypeSymbol destination,
+                XunitContext xunitContext)
+            {
+                if (destination.TypeKind == TypeKind.TypeParameter)
+                    return IsConvertibleTypeParameter(source, (ITypeParameterSymbol) destination);
+
+                var conversion = compilation.ClassifyConversion(source, destination);
+
+                if (conversion.IsNumeric)
+                    return IsConvertibleNumeric(source, destination);
+
+                // Added in https://github.com/xunit/xunit/pull/1546
+                if (xunitContext.Core.TheorySupportsConversionFromStringToDateTimesAndGuid
+                    && IsDateTimeRelatedOrGuid(destination))
+                {
+                    // Allow all conversions from strings. Narrowing conversion issues will be reported at runtime.
```

Comment:
By "narrowing conversion", do you mean whether the string is actually parsable as a date or guid?

Reply:
That's what I meant. I copied the comment from L 259, where it says about numeric conversions. I'll reword it, is `Allow all conversions from strings. All parsing issues will be reported at runtime.` acceptable?

Reply:
👍